engine/src/main/java/io/camunda/zeebe/engine/state/analyzers/CatchEventAnalyzer.java

```diff
@@ -136,12 +136,74 @@ private Either<List<DirectBuffer>, CatchEventTuple> findCatchEventInScope(
 
   public Optional<CatchEventTuple> findEscalationCatchEvent(
       final DirectBuffer escalationCode, final ElementInstance instance) {
-    // TODO walk through the scope hierarchy and look for a matching catch event
+    // walk through the scope hierarchy and look for a matching catch event
+    final var instanceRecord = instance.getValue();
+    final var process = getProcess(instanceRecord.getProcessDefinitionKey());
+
+    return findEscalationCatchEventInProcess(escalationCode, process, instance)
+        .or(
+            () -> {
+              final ElementInstance parentElementInstance =
+                  elementInstanceState.getInstance(instanceRecord.getParentElementInstanceKey());
+              if (parentElementInstance != null && parentElementInstance.isActive()) {
+                return findEscalationCatchEvent(escalationCode, parentElementInstance);
+              } else {
+                return Optional.empty();
+              }
+            });
```

Comment:
🔧 Let's add a comment to this block similar to what you had before:

`find in parent process instance if exists`